### PR TITLE
fixes #310 - Adding tests for store listing page

### LINKED ISSUE #310: Add tests for store listing page.
Add tests for store listing page.

### PR DESCRIPTION
**Pull request by djmgit**
fixes issue #310, adding tests for store listing page.

I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

### CODE REVIEW

**Comment by djmgit** [comment]
details

@Achint08 @kavithaenair @SKrPl @vibhcool please review.

File: tests/details-spec.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+describe("Testing Loklak app details page", function() {
+  beforeEach(function() {
+    browser.get("http://127.0.0.1:8080/details.html?q=MultiLinePlotter");
+  });
+
+  it("should have basic information", function() {
+    expect(element(by.css(".app-name")).getText()).toEqual("MultiLinePlotter");
+    expect(element(by.css(".app-headline")).getText()).toEqual("App to plot tweet aggregations and statistics");
```
Comment (kavithaenair) [inline tests/details-spec.js]: What if someone changes the headline or description later on?

**Comment by djmgit** [comment]
@kavithaenair Initially I though of creating a dummy app just for testing purpose, but that would not look good on the app wall (as the dummy activity would also be listed on the wall). So I decided to use MultiLinePlotter app for testing purpose. Here the main purpose is to test the controller. So if any changes are made to MultiLinePlotter app's store listing then corresponding tests have to be updated.

File: tests/details-spec.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  it("selected app category should be highlighted", function() {
+    expect(element(by.css(".selected")).getText()).toBe("Visualizer");
+  });
+
+  it("menu items should respond to click", function() {
+    let categoryIds = ["All", "Scraper", "Search", "Visualizer", "LoklakLibraries", "InternetOfThings", "Misc"];
+
+    for(let index = 0; index < categoryIds.length; index++) {
+      element(by.id(categoryIds[index])).click();
+      expect(browser.getCurrentUrl()).toBe("http://127.0.0.1:8080/#/" + categoryIds[index]);
```
Comment (vibhcool) [inline tests/details-spec.js]: This appears awesome :+1: 

I think, baseURL variable shall be used and declared at top, and baseURL shall be used everywhere in this module

**Pull request merged**